.github/workflows/release.yml

```diff
@@ -1,102 +1,415 @@
-name: Create new release
+# This file was autogenerated by dist: https://opensource.axo.dev/cargo-dist/
+#
+# Copyright 2022-2024, axodotdev
+# SPDX-License-Identifier: MIT or Apache-2.0
+#
+# CI that:
+#
+# * checks for a Git Tag that looks like a release
+# * builds artifacts with dist (archives, installers, hashes)
+# * uploads those artifacts to temporary workflow zip
+# * on success, uploads the artifacts to a GitHub Release
+#
+# Note that the GitHub Release will be created with a generated
+# title/body based on your changelogs.
 
+name: Release
+permissions:
+  "attestations": "write"
+  "contents": "write"
+  "id-token": "write"
+
+# This task will run whenever you push a git tag that looks like a version
+# like "1.0.0", "v0.1.0-prerelease.1", "my-app/0.1.0", "releases/v1.0.0", etc.
+# Various formats will be parsed into a VERSION and an optional PACKAGE_NAME, where
+# PACKAGE_NAME must be the name of a Cargo package in your workspace, and VERSION
+# must be a Cargo-style SemVer Version (must have at least major.minor.patch).
+#
+# If PACKAGE_NAME is specified, then the announcement will be for that
+# package (erroring out if it doesn't have the given version or isn't dist-able).
+#
+# If PACKAGE_NAME isn't specified, then the announcement will be for all
+# (dist-able) packages in the workspace with that version (this mode is
+# intended for workspaces with only one dist-able package, or with all dist-able
+# packages versioned/released in lockstep).
+#
+# If you push multiple tags at once, separate instances of this workflow will
+# spin up, creating an independent announcement for each one. However, GitHub
+# will hard limit this to 3 tags per commit, as it will assume more tags is a
+# mistake.
+#
+# If there's a prerelease-style suffix to the version, then the release(s)
+# will be marked as a prerelease.
 on:
+  pull_request:
```

Comment:
Why is it important to run this workflow on pull requests? Running a release workflow on a pull request is on first sight a bit confusing to me.

Reply:
Ah ok, I guess we do it because we want to catch breaking changes to dist in a PR, right? If yes, then maybe let's document this somewhere.

Reply:
this is what dist suggests, and given that we will now run ci as part of the 'plan' stage, which is also what runs on prs, this seems sensible to me. i dont feel strongly about it, but it does mean we have effectively one ci workflow, and release is just about it 'going further' than the pr workflow.

Reply:
actually, i think i will change this around a bit

Reply:
ive moved ci to the 'host' stage, so we should still run it deliberately on pr instead of relying on the plan running on PR. but i think its still good for plan to run on each pr, so we know release.yml is still valid, as they suggest.